File: Medium/2483. Minimum Penalty for a Shop.java
class Solution {
    public int bestClosingTime(String customers) {
        char[] customersToChars = customers.toCharArray();
        int minPen = 0, bestTime = 0;
        for (char c : customersToChars) {
            if (c == 'Y') {
                minPen++;
            } else {
                minPen--;
            }
        }
        int currPen = minPen;
        for (int time = 1; time <= customersToChars.length; time++) {
            if (customersToChars[time - 1] == 'Y') {
                currPen--;
            } else {
                currPen++;
            }
            if (currPen < minPen) {
                bestTime = time;
                minPen = currPen;
            }
        }
        return bestTime;
    }
}

File: Medium/2483. Minimum Penalty for a Shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        y, n = customers.count('Y'), 0
        min_penalty = y
        best_closing_time = -1
        for i, v in enumerate(customers):
            if v == 'Y':
                y -= 1
            else:
                n += 1
            curr_penalty = n + y
            if curr_penalty < min_penalty:
                min_penalty = curr_penalty
                best_closing_time = i
        return best_closing_time + 1